LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -252,7 +252,45 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            if (ClassroomDictionary.Count == 0)
+            {
+                ReportError(missingResource: "Classroom", failedAction: "Register");
+            }
+            else if (ActiveStudentList.Count == 0)
+            {
+                ReportError(missingResource: "Student", failedAction: "Register");
+            }
+            else
+            {
+                int maxCount = 0;
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        maxCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int studentCount = AttributeGenerator.random.Next(0, (classroom.MaximumCapacity - classroom.ItemList.Count - ActiveStudentList.Count) + 1);
```

Comment:
The logic how you choose the upper limit is incorrect:
You can register (classroom.MaximumCapacity - classroom.ItemList.Count) more students. If your active student count is less than (classroom.MaximumCapacity - classroom.ItemList.Count), then you have to use active student count +1 as the upper limit.

Reply:
Ok.